driver-async/src/main/com/mongodb/async/client/ChangeStreamIterable.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client;
+
+import com.mongodb.client.model.Collation;
+import com.mongodb.client.model.FullDocument;
+import org.bson.conversions.Bson;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Iterable for change streams.
+ *
+ * @param <TResult> The type of the result.
+ * @mongodb.server.release 3.6
+ * @since 3.6
+ */
+public interface ChangeStreamIterable<TResult> extends MongoIterable<TResult> {
+
+    /**
+     * Sets the fullDocument value.
+     *
+     * @param fullDocument the fullDocument
+     * @return this
+     */
+    ChangeStreamIterable<TResult> fullDocument(FullDocument fullDocument);
```

Comment:
I would have expected to be able to ask for the fullDocument, but not to be able to set it.

Same for resumeToken etc... below.